### PR TITLE
Make it possible to build fio on NetBSD

### PR DESCRIPTION
**Pull request by bvanassche**
To verify whether or not my Linux-specific changes did not cause any portability problems, I tried to build fio on a NetBSD system. The patches in this series make it possible to build fio on NetBSD. The only verification I have done of these changes is to verify that "make test" passes on both Linux and NetBSD.

### CODE REVIEW

**Comment by axboe** [comment]
Pulled, thanks Bart.

**Comment by krytarowski** [comment]
Out of curiosity, are there differences between NetBSD and Linux in terms of benchmarking results?

File: configure [hunk 1]
```diff
@@ -280,7 +282,7 @@ fi
 # cross-compiling to one of these OSes then you'll need to specify
 # the correct CPU with the --cpu option.
 case $targetos in
-AIX|OpenBSD)
+AIX|*BSD)
```
Comment (sitsofe) [inline configure]: @bvanassche Won't this also disable it on FreeBSD?

Comment (bvanassche) [reply configure]: Is FreeBSD already supported? From the configure script:

    $ grep targetos= configure 
    targetos=""
      targetos="Android"
      targetos="Linux"
      targetos='OpenBSD'
      targetos='NetBSD'
      targetos='SunOS'
      targetos='CYGWIN'
      targetos=`uname -s`

**Comment by bvanassche** [comment]
On 03/12/18 19:11, Kamil Rytarowski wrote:
> Out of curiosity, are there differences between NetBSD and Linux in 
> terms of benchmarking results?

That's a great question. Anyone who wants to know this can run the 
necessary measurements.

File: configure (hunk 1 above)
Comment (axboe) [reply configure]: FreeBSD is supported and works, we just don't check for it specifically. As such, the change would not have disabled lex on FreeBSD, but I still think the targetos check should just target netbsd and openbsd specifically.

Comment (sitsofe) [reply configure]: @bvanassche it is - it just falls into the `uname -s` case.

Comment (axboe) [reply configure]: Oh yes, good point, it is actually broken as-is.

Comment (sitsofe) [reply configure]: @axboe Thanks for fixing that one up (I'm not a FreeBSD user, just a curious bystander)!

Comment (axboe) [reply configure]: Thanks for the review, good catch!

**Pull request merged**